python/pacsign/pacsign/reader.py

```diff
@@ -264,9 +264,9 @@ def make_root_entry_dc(self, pub_key):
 
         # X size
         # TODO: Should be 0x20
-        root_entry.append_dword(len(pub_key.data[:32]))
+        root_entry.append_dword(self.sigsize)
```

Comment:
How is self.sizesize related to length of pubkey data?

Reply:
self.sigsize is the number of bytes in the hash and signature.  Either 32 or 48.  All hashes and signatures are the same length.